pkg/publish/ingress.go

```diff
@@ -21,7 +23,15 @@ import (
 
 func toPrefix(domain, serviceName string, appInstance *v1.AppInstance) (hostPrefix string) {
 	if strings.HasSuffix(domain, "on-acorn.io") {
-		appInstanceIDSegment := strings.SplitN(string(appInstance.GetUID()), "-", 2)[0]
+		appInstanceSplitName := strings.Split(appInstance.GetName()+"-"+serviceName, "-")
+		var appInstanceIDSegment string
+		var appInstanceIDSegmentByte [32]byte
+
+		for _, sliceName := range appInstanceSplitName {
+			appInstanceIDSegment += sliceName + ":"
+		}
+		appInstanceIDSegmentByte = sha256.Sum256([]byte(appInstanceIDSegment))
+		appInstanceIDSegment = hex.EncodeToString(appInstanceIDSegmentByte[:])[:12]
 		hostPrefix = name.Limit(serviceName+"-"+appInstance.GetName(), 63-len(domain)-len(appInstanceIDSegment)-1) + "-" + appInstanceIDSegment
```

Comment:
I now see that this line is actually wrong. I know you didn't introduce it, but can you fix?

The maximum length of each label is 63 characters, and a full domain name can have a maximum of 255 characters. 

So, taking len(domain) into account when determining max length is not right. Actually, to fully ensure we are compliant, we should make sure the hostPrefix is 63 chars or less and THEN make sure that hostPrefix + domain is 255 or less.